src/tabs/Tab.tsx

```diff
@@ -0,0 +1,202 @@
+ * @param {tabType} type of tabs
+ * @param {selectedTabIndex} which tab to is selected to show first
+ * @param {overflowPivot} after how many tab remaining to put in overflow menu
+ */
+type TabProp = {
+    tabs: TabPane[];
+    tabOrientation: TabOrientation;
+    tabType: TabType;
+    selectedTabIndex?: number;
+    overflowPivot?: number;
+};
+
+/**
+ * Tab Component of clarity divide content into separate views which users navigate between.
+ */
+export class Tab extends React.PureComponent<TabProp, TabState> {
+    constructor(props: TabProp) {
+        super(props);
+        this.initializeTabs(this.props.tabs);
+    }
+
+    //Initialize tabs
+    initializeTabs = (tabs: TabPane[]) => {
+        // set index in array and selected tab
+        const {tabType, selectedTabIndex} = this.props;
+        tabs.map((tab: TabPane, index: number) => {
+            tab.index = index;
+            index === (selectedTabIndex && selectedTabIndex <= tabs.length ? selectedTabIndex : 0)
+                ? (tab.isTabSelected = true)
+                : (tab.isTabSelected = false);
+            tabType == TabType.SIMPLE
+                ? (tab.disableTab = false)
+                : tab.disableTab
+                ? (tab.disableTab = true)
+                : (tab.disableTab = false);
+        });
+        this.state = {
+            tabs: [...tabs],
+            tabComponent: tabs[selectedTabIndex && selectedTabIndex <= tabs.length ? selectedTabIndex : 0].tabComponent,
+            overflowTab: {
+                tabName: "overflow",
+                tabComponent: <div />,
+                isTabSelected: false,
+                disableTab: false,
+            },
+        };
+    };
+
+    //Function handelling normal tab click
+    tabClicked = (evt: React.MouseEvent<HTMLElement>, clickedTab: TabPane) => {
+        const {tabs} = this.state;
+
+        tabs.map((tab: TabPane) => {
+            clickedTab.index === tab.index ? (tab.isTabSelected = true) : (tab.isTabSelected = false);
+        });
+        this.setState({
+            tabs: [...tabs],
+            tabComponent: clickedTab.tabComponent,
+        });
+    };
+
+    //Render Tab Button
+    private renderTab = (tab: TabPane) => {
+        let className;
+        tab.isTabSelected ? (className = ClassNames.TABACTIVE) : (className = ClassNames.TABINACTIVE);
+        return (
+            <li className={ClassNames.TABITEM}>
+                <Button className={className} disabled={tab.disableTab} onClick={evt => this.tabClicked(evt, tab)}>
+                    {tab.tabName}
+                </Button>
+            </li>
+        );
+    };
+
+    //Render Tab bar
+    private renderTabList = (tabs: TabPane[]) => {
+        const {tabType, overflowPivot} = this.props;
+        return (
+            <ul className={ClassNames.TABMAIN}>
+                {tabs.map((tab: TabPane, index: number) => {
+                    if (tabType === TabType.OVERFLOW && overflowPivot && index < overflowPivot)
```

Comment:
Reduce number of ifs.
what if index  > overflowPivot ?? 
   

}

Reply:
Done.